controllers/glance_controller.go

```diff
@@ -134,21 +134,14 @@ func (r *GlanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 	// when a condition's state doesn't change.
 	savedConditions := instance.Status.Conditions.DeepCopy()
 
-	// Always patch the instance status when exiting this function so we can persist any changes.
+	// Always patch the instance status when exiting this function so we can
+	// persist any changes.
 	defer func() {
-		// update the Ready condition based on the sub conditions
-		if instance.Status.Conditions.AllSubConditionIsTrue() {
-			instance.Status.Conditions.MarkTrue(
-				condition.ReadyCondition, condition.ReadyMessage)
-		} else {
```

Comment:
Basically all the Conditions are evaluated (as a checklist) within the main `Reconcile` or `reconcileNormal` function. The current flow that `re-init` conditions starts marking `ReadyCondition` as `False`: if the Reconcile loop, that is supposed to check all the `sub-Conditions`, reaches the end, we can Mark it as `True`, otherwise is set to `False`.
In general I think we can keep that logic in the `defer` function, from a functional point of view it's pretty much the same, but I think it probably helps people to just "follow" the logic within the main Reconcile loop.
I'm wondering if I'm overlooking something or what I did can generate a different behavior.

Reply:
Okay, thank you for explaining.  If there is functional parity, I'm good with it -- especially if we're already implementing it this way across the operators.  If we notice some odd behavior we can revisit the consideration.